tls/s2n_connection.c

```diff
@@ -231,6 +231,8 @@ struct s2n_connection *s2n_connection_new(s2n_mode mode)
      * is received, the stuffer will be resized according to the cookie length */
     GUARD_PTR(s2n_stuffer_growable_alloc(&conn->cookie_stuffer, 0));
 
+    conn->application_protocols_overridden = (struct s2n_blob){ 0 };
```

Comment:
This was fun to try and find out why the state of the `application_protocols_overridden` was nonsensical when I tried to write to it.